adventurelib.py

```diff
@@ -47,12 +53,12 @@ class Room:
     @staticmethod
     def add_direction(forward, reverse):
         """Add a direction."""
-        for dir in (forward, reverse):
-            if not dir.islower():
+        for direc in (forward, reverse):
```

Comment:
Fair enough; my IDE complained and half of what I did was just trying to line the codebase up with PEP8 (I thought PEP8 said something about shadowed names, but I could be wrong)

Reply:
> The dir() builtin is really rarely use in real code

That's so not true. You can even make stuff easier that way and even more if you go for `vars() / __dict__` e.g. with `requests` library -> `vars(requests)[get_post_whatever_str](...)` to avoid branches.